omniledger/service/service.go

```diff
@@ -930,18 +985,38 @@ clientTransactions:
 			}
 			states = append(states, scs...)
 			cin = cout
+		}
+		txsz += txrSize([]TxResult{tx})
 
+		// If the size of this one transaction alone is bigger than one block, it has to be rejected so it
+		// cannot sit on the front of the queue blocking all others.
+		if txsz > maxsz {
+			log.Errorf("%s transaction size %v is bigger than one block (%v), dropping it.", s.ServerIdentity(), txsz, maxsz)
```

Comment:
Dropping it means that the transaction will be removed from the list of available transactions? Or will it be included in the next block?

Reply:
Hmm, this is a good point. Where `createStateChanges` is called in `startPolling`, it assumes `createStateChanges` won't skip any client transactions, because it does `txs = txs[len(txOut):]`. If a transaction is skipped, it might be removed and never re-tried, and a later transaction might be replayed. Does this sound right, @jeffallen ?

Reply:
Even refused transactions go into the block, so a tx that is so big it cannot fit into a block cannot be refused, it has to be dropped entirely, with no trace (except this message in the log).

It seems like the current createStateChanges api can't support that anymore. Looking and thinking still.

Reply:
The immediate fix we can do is to prevent these from arriving at all in AddTransaction (and getTx). We can do a much bigger refactoring later, which I've added to the notes on #1409.